prog/vnet/rekey_tunnel.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::RekeyTunnel < Prog::Base
+  subject_is :nic
+
+  def start
+    ipsec_tunnel = nic.src_ipsec_tunnels.first
+    outs = ipsec_tunnel.cmd_src_nic("sudo ip -n #{ipsec_tunnel.vm_name(ipsec_tunnel.src_nic)} xfrm policy").split("\n")
+    old_spi4 = outs[outs.find_index("src #{ipsec_tunnel.src_nic.private_ipv4} dst #{ipsec_tunnel.dst_nic.private_ipv4} ")+3].split(" ")[3]
+    old_spi6 = outs[outs.find_index("src #{ipsec_tunnel.src_nic.private_ipv6} dst #{ipsec_tunnel.dst_nic.private_ipv6} ")+3].split(" ")[3]
+
+    strand.stack.last["old_spi4"] = old_spi4
```

Comment:
are you sure you mean to modify the bottom stack frame? Normally, you'd be modifying the current stack frame, which is `.first`, although modification of the stack (rather than push/pop/bud) is also pretty usual.